### DISCUSSION TITLE
How can I scroll the hover doc using `<C-f> & <C-b>`

### DISCUSSION THREAD

**Discussion by konosubakonoakua** [mrcjkb/rustaceanvim#306] [Q&A] [answered]
I can do this in a lua file which maybe a feature by `noice.nvim` (noice should take over lsp hover win, but not working with rustaceanvim).

Currently, I can only use `kk` to focus hover doc then navigate in it.

**Comment by mrcjkb** [comment, marked as answer]
Hey 👋 

I had a quick look at the noice.nvim codebase.
It looks like it [overrides the built-in LSP hover handler](https://github.com/folke/noice.nvim/blob/bf67d70bd7265d075191e7812d8eb42b9791f737/lua/noice/lsp/hover.lua#L10) so that its scrolling functions can find the window.

rustaceanvim's LSP client doesn't use the built-in handler by default (to support hover actions), so there's nothing I can do to support noice.nvim.

You can configure rustaceanvim (`:h rustaceanvim.config`) to use the built-in hover handler (in which case you lose code actions with the `K` keymap, but you can still use it with `:RustLsp hover actions`).

It might be a viable option to add a scrolling command like `:RustLsp scrollFloat [up|down]` that works with hover actions, explainError, etc, which you could then invoke via a keymap.
But I'd have to think about it.

**Reply by konosubakonoakua** [reply]
Thank U~
